### PR TITLE
Add examples for various inventory setups to the documentation

### LINKED ISSUE #12480: Write sample inventory setups
##### ISSUE TYPE
Feature Idea

##### COMPONENT NAME
core

##### ANSIBLE VERSION
2.1

##### SUMMARY
Add a few examples of different inventory setups and classifications. Capture a few styles and business needs.
As example use a per environment inventory (dev,qa,prod) with groups for location (dc1, dc2) and function (lb, web, app, db). Another can be using divisions by BU/Client, etc.

### PR DESCRIPTION
**Pull request by zerok**
Fixes: #12480.

### CODE REVIEW

File: docs/docsite/rst/user_guide/intro_inventory.rst [hunk 1]
```diff
@@ -649,6 +650,92 @@ For a full list with available plugins and examples, see :ref:`connection_plugin
...
+If you need to manage multiple environments it's sometimes prudent to
+have only hosts of a single environment defined per inventory. This
+way, it is harder to, for instance, accidentally change the state of
+nodes inside the "test" environment when you actually wanted to update
+some "staging" servers.
+
+For the example mentioned above you could have an
+:file:`inventory_test` file::
+
+  [database-servers]
```
Comment (bcoca) [inline docs/docsite/rst/user_guide/intro_inventory.rst]: avoid dashes, -servers seems redundant anyways

File: docs/docsite/rst/user_guide/intro_inventory.rst [hunk 2]
```diff
@@ -649,6 +650,92 @@ For a full list with available plugins and examples, see :ref:`connection_plugin
...
+  db01.test.example.com
+  db02.test.example.com
+
+  [app-servers]
+  app01.test.example.com
+  app02.test.example.com
+  app03.test.example.com
+
+In there are only host that are part of the "test"
```
Comment (samccann) [inline docs/docsite/rst/user_guide/intro_inventory.rst]:
```suggestion
That file only includes hosts that are part of the "test"
```

File: docs/docsite/rst/user_guide/intro_inventory.rst [hunk 3]
```diff
@@ -649,6 +650,92 @@ For a full list with available plugins and examples, see :ref:`connection_plugin
...
+environment. "Staging" machines would be defined in another file
```
Comment (samccann) [inline docs/docsite/rst/user_guide/intro_inventory.rst]:
```suggestion
environment. Define the "staging" machines in another file
```

File: docs/docsite/rst/user_guide/intro_inventory.rst [hunk 4]
```diff
@@ -649,6 +650,92 @@ For a full list with available plugins and examples, see :ref:`connection_plugin
...
+  [database-servers]
+  db01.staging.example.com
+  db02.staging.example.com
+
+  [app-servers]
+  app01.staging.example.com
+  app02.staging.example.com
+  app03.staging.example.com
+
+If you then have a playbook called :file:`site.yml`, then you'd apply
```
Comment (samccann) [inline docs/docsite/rst/user_guide/intro_inventory.rst]:
```suggestion
To apply a playbook called :file:`site.yml`
```

File: docs/docsite/rst/user_guide/intro_inventory.rst [hunk 5]
```diff
@@ -649,6 +650,92 @@ For a full list with available plugins and examples, see :ref:`connection_plugin
...
+that to all the app-servers in the test environment using the
```
Comment (samccann) [inline docs/docsite/rst/user_guide/intro_inventory.rst]:
```suggestion
to all the app-servers in the test environment, use the
```

File: docs/docsite/rst/user_guide/intro_inventory.rst [hunk 6]
```diff
@@ -649,6 +650,92 @@ For a full list with available plugins and examples, see :ref:`connection_plugin
...
+
+.. _inventory_setup-per_function:
+
+Example: Group by function
+--------------------------
+
+In the previous section you already saw an example for using groups in
+order to cluster hosts that have the same function. This allows you,
+for instance, to define firewall rules inside a playbook or role
+without affecting database servers::
```
Comment (samccann) [inline docs/docsite/rst/user_guide/intro_inventory.rst]: please use code-block here

**Comment by acozine** [comment]
Thanks @zerok for submitting these examples. We appreciate the time and effort you've put in. If you're interested in documentation, please consider joining the [Docs Working Group](https://github.com/ansible/community/wiki/Docs)! 

Can you take a look at the comments above? If you have questions, need clarification, etc., feel free to reach out on the #ansible-docs channel on freenode IRC.

**Comment by zerok** [comment]
Thank you 😊I will try to find time for implementing your comments tomorrow 🙂

File: docs/docsite/rst/user_guide/intro_inventory.rst (hunk 1 above)
Comment (zerok) [reply docs/docsite/rst/user_guide/intro_inventory.rst]: Fixed :) The "servers" postfix is consistent with other group-names in that document, though, so I opted to keep it.

File: docs/docsite/rst/user_guide/intro_inventory.rst (hunk 6 above)
Comment (zerok) [reply docs/docsite/rst/user_guide/intro_inventory.rst]: Fixed :)

File: docs/docsite/rst/user_guide/intro_inventory.rst [hunk 7]
```diff
@@ -649,6 +650,100 @@ For a full list with available plugins and examples, see :ref:`connection_plugin
...
+  [appservers]
+  app01.staging.example.com
+  app02.staging.example.com
+  app03.staging.example.com
+
+To apply a playbook called :file:`site.yml`
+to all the app servers in the test environment, use the
+following command::
+
+  ansible-playbook -i inventory_test site.yml -l app-servers
```
Comment (samccann) [inline docs/docsite/rst/user_guide/intro_inventory.rst]:
```suggestion
  ansible-playbook -i inventory_test site.yml -l appservers
```

**Review by samccann — COMMENTED** [review]
leftover nit from removing the dash from the groups :-)

File: docs/docsite/rst/user_guide/intro_inventory.rst (hunk 7 above)
Comment (zerok) [reply docs/docsite/rst/user_guide/intro_inventory.rst]: 🙈😂

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
Thanks @zerok for the inventory examples!

**Comment by zerok** [comment]
Thank you 😊

**Pull request merged**